### PR TITLE
HTTPClient refactor

### PR DESCRIPTION
**Pull request by richardm-stripe**
## Summary
Contains refactoring changes from https://github.com/stripe/stripe-python/pull/1165 that we can merge into master to minimize how much master and beta diverge.

In particular, it extracts an `HTTPClientBase` parent class from `HTTPClient`, and removes some dead code related to warning that would only fire if Python2 (which this library is no longer compatible with) is used.

### CODE REVIEW

File: flake8_stripe/flake8_stripe.py [hunk 1]
```diff
@@ -158,6 +160,7 @@ def run(self) -> Iterator[Tuple[int, int, str, type]]:
                     )
             if isinstance(node, ast.ImportFrom):
                 # Forbid: from stripe...module import Type
+                assert node.module is not None
```
Comment (pakrym-stripe) [inline flake8_stripe/flake8_stripe.py:163]: hm, why?

File: stripe/_http_client.py [hunk 2]
```diff
@@ -315,16 +311,8 @@ def close(self):
...
-        self,
-        timeout: int = 80,
-        session: Optional["Session"] = None,
-        verify_ssl_certs: bool = True,
-        proxy: Optional[Union[str, HTTPClient._Proxy]] = None,
-    ):
-        super(RequestsClient, self).__init__(
-            verify_ssl_certs=verify_ssl_certs, proxy=proxy
-        )
+    def __init__(self, timeout=80, session=None, **kwargs):
```
Comment (pakrym-stripe) [inline stripe/_http_client.py:314]: types lost

Comment (pakrym-stripe) [reply stripe/_http_client.py:314]: here and else where

**Review by pakrym-stripe — APPROVED** [review]

File: flake8_stripe/flake8_stripe.py (hunk 1 above)
Comment (richardm-stripe) [reply flake8_stripe/flake8_stripe.py:163]: Fixed a pyright info I think. I can remove.

**Pull request merged**